src/App.jsx

```diff
@@ -145,16 +145,20 @@ function App() {
         return setContext(initContext(context));
       }
       if (mode === "cloud") {
-        project(id)
-          .then((result) => {
-            initVfs(result);
-            return setContext(initContext(result));
-          })
-          .catch(() => {
-            progressElement.classList.add("hidden");
-
-            return setContext("error");
-          });
+        project(id).then((result) => {
+          initVfs(result);
+          return setContext(initContext(result));
+        });
+      }
+      if (mode === "chat") {
+        const context = sampleProject();
+        initVfs(context);
+        return setContext(initContext(context));
+      }
+      if (mode === "mobile") {
+        return setContext("mobile");
+      } else {
+        window.location.assign(`${window.location.origin}/ide/sample/api`);
```

Comment:
Can we also do this with router redirect instead?

Reply:
We cannot use react-router-dom in components that are not wrapped with Route. Therefore I used pure js.

Here we select the mode for the context, if the mode is wrong, the application cannot switch to RouteManager because the context cannot be created.

Then we can solve this by lowering the initMode and initContext functions to lower levels.